### ISSUE TITLE
About the trainning speed.

### ISSUE THREAD

**Issue by Zakeiswo** [ziwei-zh/CorrPM#3]
It's weird for my PC to train the code, it's very slow, can you figure it out.
In order to use PyTorch:1.4.0, I replace the "InPlaceABNSync" with normal "nn.BatchNorm2d".

CUDA:10.0
PyTorch:1.4.0
RTX 2080ti 11G
Ubuntu 16.04
i7-9700k
RAM：32 GB
`iter:0/7616,loss:8.788,lr:1.000e-03,time:8.2
iter:500/7616,loss:3.455,lr:9.997e-04,time:165.6
iter:1000/7616,loss:2.483,lr:9.993e-04,time:166.3
iter:1500/7616,loss:3.580,lr:9.990e-04,time:167.0
iter:2000/7616,loss:2.938,lr:9.987e-04,time:167.5
iter:2500/7616,loss:2.709,lr:9.984e-04,time:166.9
iter:3000/7616,loss:3.379,lr:9.980e-04,time:167.1
iter:3500/7616,loss:2.261,lr:9.977e-04,time:164.4
iter:4000/7616,loss:2.764,lr:9.974e-04,time:164.3
iter:4500/7616,loss:3.147,lr:9.970e-04,time:164.3
iter:5000/7616,loss:3.124,lr:9.967e-04,time:164.3
iter:5500/7616,loss:2.995,lr:9.964e-04,time:164.3
iter:6000/7616,loss:1.715,lr:9.961e-04,time:164.4
iter:6500/7616,loss:2.305,lr:9.957e-04,time:164.0
iter:7000/7616,loss:1.912,lr:9.954e-04,time:165.5
iter:7500/7616,loss:1.704,lr:9.951e-04,time:165.5
` 
Thank you.

**Comment by ziwei-zh** [comment]
I think it is normal because there are many matrix multiplications in HNL.

**Comment by Zakeiswo** [comment]
Thx，you are so kind. I will try another way.

**Comment by Zakeiswo** [comment]
Sorry to bother you, when I removed the HNL and doubled the batch_size(from 8 to 16), it's still very slow. Could you give me some ideas to increase the speed？For example，delete supervise from the edge part, or anything else.

iter:0/3808,loss:8.330,lr:1.000e-03,time:7.1
iter:500/3808,loss:2.914,lr:9.993e-04,time:234.0
iter:1000/3808,loss:2.611,lr:9.987e-04,time:235.3
